### ISSUE TITLE
[Dashboard] Dashboard fails to build

### ISSUE THREAD

**Issue by peytondmurray** [ray-project/ray#30871]
### What happened + What you expected to happen

The Ray dashboard currently (f7a29503a) fails to build:

```
[~/Desktop/workspace/ray/dashboard/client]⟦ray⟧[update-logging-config]$ yarn build
yarn run v1.22.19
$ react-scripts build
Creating an optimized production build...
Failed to compile.

TS2769: No overload matches this call.
  Overload 1 of 2, '(props: { component: ElementType<any>; } & { children?: ReactNode; color?: "inherit" | "error" | "disabled" | "primary" | "secondary" | "action" | undefined; fontSize?: "small" | ... 4 more ... | undefined; htmlColor?: string | undefined; shapeRendering?: string | undefined; titleAccess?: string | undefined; viewBox?: string | undefined; } & CommonProps<...> & Pick<...>): Element', gave the following error.
    Property 'component' is missing in type '{ onClick: MouseEventHandler<Element>; }' but required in type '{ component: ElementType<any>; }'.
  Overload 2 of 2, '(props: DefaultComponentProps<SvgIconTypeMap<{}, "svg">>): Element', gave the following error.
    Type 'MouseEventHandler<Element>' is not assignable to type 'MouseEventHandler<SVGSVGElement>'.
      Types of parameters 'event' and 'event' are incompatible.
        Type 'MouseEvent<SVGSVGElement, MouseEvent>' is missing the following properties from type 'MouseEvent<Element, MouseEvent>': detail, view
    12 |
    13 | export const Minimizer: React.FC<MinimizerProps> = ({ onClick }) => (
  > 14 |   <ExpandLessIcon onClick={onClick} />
       |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    15 | );
    16 |
    17 | export const Expander: React.FC<ExpanderProps> = ({ onClick }) => (


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Versions / Dependencies

Ray version: f7a29503a
Python: 3.10.8
OS: Linux 6.0.10-arch2-1 x86_64

### Reproduction script

```bash
cd dashboard/client
yarn
yarn build
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by richardliaw** [comment]
@rkooo567 @alanwguo take a look when you get the chance?

**Comment by alanwguo** [comment]
We only guarantee support for `npm` for the dashboard development. Although I wouldn't expect `yarn` to fail.

Can you try running `npm ci && npm install`?

**Comment by alanwguo** [comment]
Can you also try deleting your `node_modules` folder before running `yarn` or `npm ci`? I suspect there is some package version issues in your environment.

**Comment by peytondmurray** [comment]
Hmm, doing a `git clean -fdx` to clear out any build stuff didn't seem to work when I tried `yarn && yarn build` afterward. But doing `git clean -fdx && npm ci && npm run build` did seem to work, so it must be a dependency that has been updated since the lockfile was committed, and `yarn` must have been pulling a newer version. Thanks for the help, marking this issue as resolved! FWIW I'd definitely like to see yarn supported, builds are _much_ faster with it.
